### DISCUSSION TITLE
Not able to get an event from API Gateway in my Mockoon container

### DISCUSSION THREAD

**Discussion by bhatnagarm** [mockoon/mockoon#1432] [Help] [answered]
**I am not able to get the event coming from API Gateway**

I'm my current operation I want to get the API Gateway event and use a custom header to find the key to my S3 bucket. I have tried a lot of ways and all of them are failing on return.

**Steps to Reproduce**
* Create a JavaScript Lambda with Mockoon Serverless

```
import { S3Client, HeadObjectCommand, GetObjectCommand } from "@aws-sdk/client-s3";
import {MockoonServerless} from '@mockoon/serverless';

const client = new S3Client({});
const bucket = process.env.DEFINITIONS_BUCKET;
const key = 'mackoon-cli-environment.json';

const { ContentType } = await client.send(new HeadObjectCommand({
    Bucket: bucket,
    Key: key,
}));

const s3Object = await client.send(new GetObjectCommand({
	Key: key,
	Bucket: bucket
}));

console.log('CONTENT TYPE:', ContentType);
const mockEnv = JSON.parse(await s3Object.Body.transformToString());
const mockoonServerless = new MockoonServerless(mockEnv, {
    logTransaction: true,
    envVarsPrefix: '',
    enableAdminApi: true,
    disableTls: true
});

export const handler = async (event) => {
  console.log("EVENT: " + event.headers['x-source-location']);
  return await new Promise(handler => mockoonServerless.awsHandler());
};
// export const handler = mockoonServerless.awsHandler();
```

The response I get for this code with API Gateway
```
Test Event Name
API_Gateway_Sample

Response
null

Function Logs
2024-06-27T00:30:55.266Z	undefined	INFO	CONTENT TYPE: application/json
START RequestId: a3c06028-30a2-4fe1-8416-3a180db3b093 Version: $LATEST
2024-06-27T00:30:55.273Z	a3c06028-30a2-4fe1-8416-3a180db3b093	INFO	EVENT: int-process-api-electrical-work-request-dev/v1
END RequestId: a3c06028-30a2-4fe1-8416-3a180db3b093
REPORT RequestId: a3c06028-30a2-4fe1-8416-3a180db3b093	Duration: 83.13 ms	Billed Duration: 84 ms	Memory Size: 256 MB	Max Memory Used: 204 MB	Init Duration: 2841.79 ms

```

If I comment out the above and just use the one suggested in documentation it will work.  Is there a good way for me to get the events and then execute the Mockoon Serverless application.

**Mockoon version:**

All Versions

**Comment by 255kb** [comment]
I never tried using an async handler, but wouldn't this be enough?

```js
export const handler = async (event) => {
  console.log("EVENT: " + event.headers['x-source-location']);
  return mockoonServerless.awsHandler();
};
```

**Comment by bhatnagarm** [comment, marked as answer]
Thanks @255kb for your response,
Apologies I didn't return earlier. 
The code suggested above didn't work for me. I think awsHandler() was having issues within Async block. 
This workaround worked well though for me:

```
export const handler = async (event) => {
console.log("EVENT: " + event.headers['x-source-location']);
return await mockoonServerless.awsHandler()(event);
};
```

I think this might be a useful update to the documentation to add this block of code for people who want to use it. Happy to provide a MR with the changes.

**Reply by 255kb** [reply]
Great to know it works! 
Feel free to open a PR to update the serverless package's readme. 
Thank you.

**Comment by bhatnagarm** [comment]
Hi @255kb , 
I have tried to push my feature branch into the project but seems I don't have access to create the feature.

**Reply by 255kb** [reply]
Hi, you won't be able to push to this repo as you don't have the rights. 
You forked it, so you must create a branch on your fork, commit and push. Then you [open a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) on this repository, asking for a merge of your fork's branch.
